logic/save_logic.py

```diff
@@ -644,14 +644,27 @@ def get_additional_parameters(self):
         """ Method that return the additional parameters dictionary securely """
         return self._additional_parameters.copy()
 
-    def update_additional_parameters(self, **new_pairs):
-        """ Method to update one or multiple additional parameters """
-        dic = {}
-        for key in new_pairs.keys():
-            dic[key] = netobtain(new_pairs[key])
-        self._additional_parameters = {**self._additional_parameters, **dic}
+    def update_additional_parameters(self, __pdict=None, **new_params):
```

Comment:
Well since positional-only arguments are not yet supported (https://www.python.org/dev/peps/pep-0570/) I wanted to avoid any naming conflicts with keyword arguments.
Any suggestions for a better, highly unlikely keyword?
The alternative would be to use `*args` and check for `len(args) < 2` to raise an exception.

Reply:
Why do you need to pass a dict in the first place?
You can just unwrap it with ** and only use new_params, if I understand the code correctly.

Reply:
Well in principle you are right but the additional_parameter names (i.e. dict keys) must be valid argument keywords then. This restricts parameter name format for example if you want to use blanks in your name. 
Since they are just written into a text file it would be good to accept all strings.

Reply:
Then why does **new_params exist?

Reply:
Well first of all this is how the author first implemented it and second I personally find it nice to have the choice between wrapping everything in a dict and then call the method or just giving kwargs.
I implemented similar methods in pulsed to set parameters belonging to a certain parameter group (e.g. `PulsedMeasurementLogic.set_measurement_settings`). I had jupyter notebooks in mind where it increases code readability and is more intuitive if you can just go like:
```python
set_measurement_settings(invoke_settings=True)
```
instead of
```python
set_measurement_settings({'invoke_settings'=True})
```
 It's just more versatile/convenient. 
If this is becoming a problem we can ditch this concept entirely.

Reply:
As long as Python does not support positional-only arguments, supporting both approaches is indeed a problem.

Reply:
Any objections against the `*args` approach and manually raising an exception if `len(args) > 1`?